index.js

```diff
@@ -2,33 +2,37 @@
 const url = require('url')
 const jwt = require('jsonwebtoken')
 
-module.exports = exports = (secret, whitelist) => (fn) => {
-       
+module.exports = exports = (secret, whitelist, config = {}) => (fn) => {
+
     if (!secret) {
         throw Error('micro-jwt-auth must be initialized passing a secret to decode incoming JWT token')
     }
-    
+
+    if (!Array.isArray(whitelist)) {
+        config = whitelist || {}
+    }
+
```

Comment:
yeah, I think whis is a good idea, I felt it coming when first introduced the whitelist parameter, but didn't wanted to overdue back then, i thinkg now it makes way more sense to have an options object